controllers/controller.go

```diff
@@ -0,0 +1,117 @@
+package controllers
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/darahayes/go-boom"
+	"github.com/gorilla/mux"
+	"github.com/gror/models"
+	"github.com/gror/services"
+	"gopkg.in/mgo.v2/bson"
+)
+
+func respondWithError(w http.ResponseWriter, code int, msg string) {
+	respondWithJson(w, code, map[string]string{"error": msg})
+}
+
+func respondWithJson(w http.ResponseWriter, code int, payload interface{}) {
+	response, err1 := json.Marshal(payload)
+	if err1 != nil {
+		log.Fatal(err1)
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(code)
+	w.Write(response)
+}
+func CreateDockerConfig(w http.ResponseWriter, r *http.Request) {
+
+	var rootobject model.Root
+	err := json.NewDecoder(r.Body).Decode(&rootobject)
+	if err != nil {
+		boom.BadData(w, "Unprocessable Entity error")
+		return
+	}
+
+	rootobject.ID = bson.NewObjectId()
+	marshalData, err := json.Marshal(rootobject)
+	if err != nil {
+		log.Fatal(err)
```

Comment:
We don't want our application to exit just because there was invalid data. Why can't we return respective http code with description?